src/txn/txn_recover.c

```diff
@@ -959,6 +975,14 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
 done:
     /* Close cached cursors, rollback-to-stable asserts exclusive access. */
     WT_ERR(__recovery_close_cursors(&r));
+#ifndef WT_STANDALONE_BUILD
+    if (conn->unclean_shutdown && conn->recovery_major == 10 && conn->recovery_minor == 0 &&
+      conn->recovery_patch == 0) {
+        WT_ERR_MSG(session, WT_ERROR,
+          "Upgrading a version on a database that was not shutdown cleanly is not allowed. Perform "
```

Comment:
Since this is a very specific situation, I'd suggest making the message itself very specific:
"Upgrading from a WiredTiger version 10.0.0 database that was not shutdown cleanly is not allowed. Perform a clean shutdown on version 10.0.0 and then upgrade."

Reply:
Someone else ought to think about this also, since WiredTiger versions and MongoDB versions don't align. But from a WiredTiger-centric perspective, it gives us the information we need since we know WT10.0.0 is the version we need to worry about.

Reply:
Thanks, along with the provided message, I added another sentence "WiredTiger version 10.0.0 maps to a couple of releases of MongoDB." It is just to make sure that we are not confusing the MongoDB users with the WiredTiger version. Is that helps?

Reply:
I feel adding WT standalone version in the message will confuse MDB users, and mentioning "10.0.0 maps to a couple of MDB releases" would not help (if not worse) without providing the detailed mapping information. 

My preference is to avoid mentioning the WT version in the message. We can always figure out the version info in the code logic when needed.

Reply:
WT 10.0.0 is mapped only to MongoDB 4.4.0 to 4.4.6 version only. MongoDB 5.0.0 has WT 10.0.1. Do we mention the MongoDB versions explicitly or just revert back to message without version numbers?